[FEAT] Render the Label Font of a Edge

**Is your feature request related to a problem? Please describe.**
All elements from BPMN can have a LabelStyle, who define the position, size, style... of their label. 

**Describe the solution you'd like**
We want to render the BPMN Label Font of a Edge to MxGraph.

Relates #101

With `style += ';' + this.mxConstants.STYLE_FONTSTYLE + '=' + this.mxConstants.FONT_STRIKETHROUGH;`, I have the following error, but `FONT_STRIKETHROUGH` is un the type definition of `mxConstants`: 

`Error: /Users/csouchet/Source/Process Analytics/BPMN-Js/src/component/mxgraph/MxGraphRenderer.ts(83,79): semantic error TS2339: Property 'FONT_STRIKETHROUGH' does not exist on type '{ DEFAULT_HOTSPOT: number; MIN_HOTSPOT_SIZE: number; MAX_HOTSPOT_SIZE: number; RENDERING_HINT_EXACT: string; RENDERING_HINT_FASTER: string; RENDERING_HINT_FASTEST: string; DIALECT_SVG: string; ... 259 more ...; PERIMETER_TRIANGLE: string; }'.`

--- a/src/component/mxgraph/MxGraphRenderer.ts
+++ b/src/component/mxgraph/MxGraphRenderer.ts
@@ -21,6 +21,7 @@ import ShapeBpmnElement, { ShapeBpmnEvent } from '../../model/bpmn/shape/ShapeBp
 import { MxGraphFactoryService } from '../../service/MxGraphFactoryService';
 import Waypoint from '../../model/bpmn/edge/Waypoint';
 import { StyleConstant } from './StyleConfigurator';
+import { Font } from '../../model/bpmn/Label';
 
 interface Coordinate {
   x: number;
@@ -72,41 +73,41 @@ export default class MxGraphRenderer {
   }
 
   private computeStyle(bpmnCell: Shape | Edge): string {
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    const styleValues = new Map<string, any>();
+
+    const font = bpmnCell.label?.font;
+    if (font) {
+      styleValues.set(this.mxConstants.STYLE_FONTFAMILY, font.name);
+      styleValues.set(this.mxConstants.STYLE_FONTSIZE, font.size);
+      styleValues.set(this.mxConstants.STYLE_FONTSTYLE, this.getFontStyleValue(font));
+    }
+
     const bpmnElement = bpmnCell.bpmnElement;
+    if (bpmnElement instanceof ShapeBpmnEvent) {
+      styleValues.set(StyleConstant.BPMN_STYLE_EVENT_KIND, bpmnElement.eventKind);
+    }
 
     let style = bpmnElement.kind as string;
-    if (bpmnElement instanceof ShapeBpmnEvent) {
-      style += ';' + StyleConstant.BPMN_STYLE_EVENT_KIND + '=' + bpmnElement.eventKind;
-    }
-    style += this.computeFontStyle(bpmnCell);
+    styleValues.forEach((value, key) => {
+      if (value) {
+        style += ';' + key + '=' + value;
+      }
+    });
 
     return style;
   }
 
-  private computeFontStyle(bpmnCell: Shape | Edge): string {
-    let style = '';
-
-    const font = bpmnCell?.label?.font;
-    if (font) {
-      if (font.name) {
-        style += ';' + this.mxConstants.STYLE_FONTFAMILY + '=' + font.name;
-      }
-
-      if (font.size) {
-        style += ';' + this.mxConstants.STYLE_FONTSIZE + '=' + font.size;
-      }
-
-      if (font.isBold) {
-        style += ';' + this.mxConstants.STYLE_FONTSTYLE + '=' + this.mxConstants.FONT_BOLD;
-      } else if (font.isItalic) {
-        style += ';' + this.mxConstants.STYLE_FONTSTYLE + '=' + this.mxConstants.FONT_ITALIC;
-      } else if (font.isStrikeThrough) {
-        style += ';' + this.mxConstants.STYLE_FONTSTYLE + '=' + this.mxConstants.FONT_STRIKETHROUGH;
-      } else if (font.isUnderline) {
-        style += ';' + this.mxConstants.STYLE_FONTSTYLE + '=' + this.mxConstants.FONT_UNDERLINE;
-      }
+  private getFontStyleValue(font: Font): number {
+    if (font.isBold) {
+      return this.mxConstants.FONT_BOLD;
+    } else if (font.isItalic) {
+      return this.mxConstants.FONT_ITALIC;
+    } else if (font.isStrikeThrough) {
+      return this.mxConstants.FONT_STRIKETHROUGH;
+    } else if (font.isUnderline) {
+      return this.mxConstants.FONT_UNDERLINE;
     }
-    return style;
   }
 
   private insertEdges(edges: Edge[]): void {

--- a/src/component/mxgraph/MxGraphRenderer.ts
+++ b/src/component/mxgraph/MxGraphRenderer.ts
@@ -89,11 +89,23 @@ export default class MxGraphRenderer {
     }
 
     let style = bpmnElement.kind as string;
-    styleValues.forEach((value, key) => {
-      if (value) {
-        style += ';' + key + '=' + value;
-      }
-    });
+    // styleValues.forEach((value, key) => {
+    //   if (value) {
+    //     style += ';' + key + '=' + value;
+    //   }
+    // });
+
+    style +=
+      ';' + // TODO only if the style values generate not empty string
+      [...styleValues]
+        .filter(([, v]) => v)
+        .map(([key, value]) => key + '=' + value)
+        .join(';');
+
+    // TODO refactor to manage empty string
+    // let styleArray: string[] = [bpmnElement.kind];
+    // push styleValues as string[] in styleArray
+    // join everything with ';'
 
     return style;
   }

--- a/src/component/mxgraph/MxGraphRenderer.ts
+++ b/src/component/mxgraph/MxGraphRenderer.ts
@@ -66,19 +66,19 @@ export default class MxGraphRenderer {
       const bounds = shape.bounds;
       const parent = this.getParent(bpmnElement);
       const absoluteCoordinate = { x: bounds.x, y: bounds.y };
-      const style = this.computeStyleName(shape);
+      const style = this.computeStyle(shape);
       this.insertVertexGivenAbsoluteCoordinates(parent, bpmnElement.id, bpmnElement.name, absoluteCoordinate, bounds.width, bounds.height, style);
     }
   }
 
-  private computeStyleName(shape: Shape): string {
-    const bpmnElement = shape.bpmnElement;
+  private computeStyle(bpmnCell: Shape | Edge): string {
+    const bpmnElement = bpmnCell.bpmnElement;
 
     let style = bpmnElement.kind as string;
     if (bpmnElement instanceof ShapeBpmnEvent) {
       style += ';' + StyleConstant.BPMN_STYLE_EVENT_KIND + '=' + bpmnElement.eventKind;
     }
-    style += this.computeFontStyle(shape);
+    style += this.computeFontStyle(bpmnCell);
 
     return style;
   }
@@ -114,14 +114,10 @@ export default class MxGraphRenderer {
       const bpmnElement = edge.bpmnElement;
       if (bpmnElement) {
         const parent = this.graph.getDefaultParent();
-        const mxEdge = this.graph.insertEdge(
-          parent,
-          bpmnElement.id,
-          bpmnElement.name,
-          this.getCell(bpmnElement.sourceRefId),
-          this.getCell(bpmnElement.targetRefId),
-          bpmnElement.kind,
-        );
+        const source = this.getCell(bpmnElement.sourceRefId);
+        const target = this.getCell(bpmnElement.targetRefId);
+        const style = this.computeStyle(edge);
+        const mxEdge = this.graph.insertEdge(parent, bpmnElement.id, bpmnElement.name, source, target, style);
         this.insertWaypoints(edge.waypoints, mxEdge);
       }
     });

--- a/src/component/mxgraph/MxGraphRenderer.ts
+++ b/src/component/mxgraph/MxGraphRenderer.ts
@@ -73,8 +73,7 @@ export default class MxGraphRenderer {
   }
 
   computeStyle(bpmnCell: Shape | Edge): string {
-    /* eslint-disable @typescript-eslint/no-explicit-any */
-    const styleValues = new Map<string, any>();
+    const styleValues = new Map<string, string | number>();
 
     const font = bpmnCell.label?.font;
     if (font) {
@@ -88,26 +87,9 @@ export default class MxGraphRenderer {
       styleValues.set(StyleConstant.BPMN_STYLE_EVENT_KIND, bpmnElement.eventKind);
     }
 
-    let style = bpmnElement.kind as string;
-    styleValues.forEach((value, key) => {
-      if (value) {
-        style += ';' + key + '=' + value;
-      }
-    });
-
-    // style +=
-    //   ';' + // TODO only if the style values generate not empty string
-    //   [...styleValues]
-    //     .filter(([, v]) => v)
-    //     .map(([key, value]) => key + '=' + value)
-    //     .join(';');
-
-    // TODO refactor to manage empty string
-    // let styleArray: string[] = [bpmnElement.kind];
-    // push styleValues as string[] in styleArray
-    // join everything with ';'
-
-    return style;
+    return [bpmnElement.kind as string] //
+      .concat([...styleValues].filter(([, v]) => v).map(([key, value]) => key + '=' + value))
+      .join(';');
   }
 
   private getFontStyleValue(font: Font): number {

--- a/src/component/mxgraph/MxGraphRenderer.ts
+++ b/src/component/mxgraph/MxGraphRenderer.ts
@@ -72,7 +72,7 @@ export default class MxGraphRenderer {
     }
   }
 
-  private computeStyle(bpmnCell: Shape | Edge): string {
+  computeStyle(bpmnCell: Shape | Edge): string {
     /* eslint-disable @typescript-eslint/no-explicit-any */
     const styleValues = new Map<string, any>();
 
@@ -89,18 +89,18 @@ export default class MxGraphRenderer {
     }
 
     let style = bpmnElement.kind as string;
-    // styleValues.forEach((value, key) => {
-    //   if (value) {
-    //     style += ';' + key + '=' + value;
-    //   }
-    // });
+    styleValues.forEach((value, key) => {
+      if (value) {
+        style += ';' + key + '=' + value;
+      }
+    });
 
-    style +=
-      ';' + // TODO only if the style values generate not empty string
-      [...styleValues]
-        .filter(([, v]) => v)
-        .map(([key, value]) => key + '=' + value)
-        .join(';');
+    // style +=
+    //   ';' + // TODO only if the style values generate not empty string
+    //   [...styleValues]
+    //     .filter(([, v]) => v)
+    //     .map(([key, value]) => key + '=' + value)
+    //     .join(';');
 
     // TODO refactor to manage empty string
     // let styleArray: string[] = [bpmnElement.kind];

--- a/src/component/mxgraph/StyleConfigurator.ts
+++ b/src/component/mxgraph/StyleConfigurator.ts
@@ -185,7 +185,7 @@ export default class StyleConfigurator {
     this.configureCommonDefaultStyle(style);
   }
 
-  private configureCommonDefaultStyle(style: any) {
+  private configureCommonDefaultStyle(style: any): void {
     style[this.mxConstants.STYLE_FONTSIZE] = 12;
     style[this.mxConstants.STYLE_FONTCOLOR] = 'Black';
     style[this.mxConstants.STYLE_FILLCOLOR] = 'White';

--- a/src/component/mxgraph/StyleConfigurator.ts
+++ b/src/component/mxgraph/StyleConfigurator.ts
@@ -94,12 +94,7 @@ export default class StyleConfigurator {
 
   private configureDefaultVertexStyle(): void {
     const style = this.getDefaultVertexStyle();
-    style[this.mxConstants.STYLE_HORIZONTAL] = true;
-    style[this.mxConstants.STYLE_FONTSIZE] = 12;
-    style[this.mxConstants.STYLE_FILLCOLOR] = 'white';
-    style[this.mxConstants.STYLE_FONTCOLOR] = 'black';
-    style[this.mxConstants.STYLE_STROKECOLOR] = 'black';
-    style[this.mxConstants.STYLE_LABEL_BACKGROUNDCOLOR] = 'none';
+    this.configureCommonDefaultStyle(style);
   }
 
   private configurePoolStyle(): void {
@@ -182,16 +177,20 @@ export default class StyleConfigurator {
     style[this.mxConstants.STYLE_EDGE] = this.mxConstants.EDGESTYLE_SEGMENT;
     style[this.mxConstants.STYLE_ENDARROW] = this.mxConstants.ARROW_BLOCK_THIN;
     style[this.mxConstants.STYLE_ENDSIZE] = 12;
-    style[this.mxConstants.STYLE_STROKECOLOR] = 'Black';
     style[this.mxConstants.STYLE_STROKEWIDTH] = 1.5;
     style[this.mxConstants.STYLE_ROUNDED] = 1;
     style[this.mxConstants.STYLE_ARCSIZE] = 5;
-
-    style[this.mxConstants.STYLE_FONTSIZE] = 12;
-    style[this.mxConstants.STYLE_FILLCOLOR] = 'White';
-    style[this.mxConstants.STYLE_FONTCOLOR] = 'Black';
-    style[this.mxConstants.STYLE_LABEL_BACKGROUNDCOLOR] = 'none';
     style[this.mxConstants.STYLE_VERTICAL_ALIGN] = 'bottom';
+
+    this.configureCommonDefaultStyle(style);
+  }
+
+  private configureCommonDefaultStyle(style: any) {
+    style[this.mxConstants.STYLE_FONTSIZE] = 12;
+    style[this.mxConstants.STYLE_FONTCOLOR] = 'Black';
+    style[this.mxConstants.STYLE_FILLCOLOR] = 'White';
+    style[this.mxConstants.STYLE_STROKECOLOR] = 'Black';
+    style[this.mxConstants.STYLE_LABEL_BACKGROUNDCOLOR] = 'none';
   }
 
   private configureSequenceFlowsStyle(): void {

--- a/src/component/mxgraph/StyleConfigurator.ts
+++ b/src/component/mxgraph/StyleConfigurator.ts
@@ -187,7 +187,7 @@ export default class StyleConfigurator {
     style[this.mxConstants.STYLE_ROUNDED] = 1;
     style[this.mxConstants.STYLE_ARCSIZE] = 5;
 
-    style[this.mxConstants.STYLE_FONTSIZE] = 15;
+    style[this.mxConstants.STYLE_FONTSIZE] = 12;
     style[this.mxConstants.STYLE_FILLCOLOR] = 'White';
     style[this.mxConstants.STYLE_FONTCOLOR] = 'Black';
     style[this.mxConstants.STYLE_LABEL_BACKGROUNDCOLOR] = 'none';

--- a/src/component/parser/json/converter/DiagramConverter.ts
+++ b/src/component/parser/json/converter/DiagramConverter.ts
@@ -140,17 +140,15 @@ export default class DiagramConverter extends AbstractConverter<BpmnModel> {
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   private deserializeLabel(bpmnLabel: any, id: string): Label {
-    if (bpmnLabel) {
-      const labelStyle = bpmnLabel.labelStyle;
-      if (labelStyle) {
-        const font = this.findFont(labelStyle);
+    const labelStyle = bpmnLabel?.labelStyle;
+    if (labelStyle) {
+      const font = this.findFont(labelStyle);
 
-        if (font) {
-          return new Label(font);
-        } else {
-          // TODO error management
-          console.warn('Unable to assign font %s to shape/edge %s', labelStyle, id);
-        }
+      if (font) {
+        return new Label(font);
+      } else {
+        // TODO error management
+        console.warn('Unable to assign font %s to shape/edge %s', labelStyle, id);
       }
     }
   }

--- a/test/e2e/mxGraph.model.test.ts
+++ b/test/e2e/mxGraph.model.test.ts
@@ -182,7 +182,7 @@ describe('mxGraph model', () => {
             <bpmndi:BPMNEdge bpmnElement="default_sequence_flow_id" id="E1373649849864_default_sequence_flow_id">
                 <di:waypoint x="342.0" y="351.0"/>
                 <di:waypoint x="390.0" y="351.0"/>
-                <bpmndi:BPMNLabel labelStyle="LS1373649849858">
+                <bpmndi:BPMNLabel labelStyle="bold_font_id">
                     <dc:Bounds height="12.804751171875008" width="94.93333333333335" x="153.67766754457273" y="371.3333333333333"/>
                 </bpmndi:BPMNLabel>
             </bpmndi:BPMNEdge>
@@ -352,7 +352,7 @@ describe('mxGraph model', () => {
     expectModelContainsShape('inclusiveGateway_1', ShapeBpmnElementKind.GATEWAY_INCLUSIVE);
 
     // sequence flow
-    expectModelContainsEdge('default_sequence_flow_id', SequenceFlowKind.DEFAULT, MarkerConstant.ARROW_DASH, expectedFont);
+    expectModelContainsEdge('default_sequence_flow_id', SequenceFlowKind.DEFAULT, MarkerConstant.ARROW_DASH, expectedBoldFont);
     expectModelContainsEdge('normal_sequence_flow_id', SequenceFlowKind.NORMAL);
     expectModelContainsEdge('conditional_sequence_flow_from_activity_id', SequenceFlowKind.CONDITIONAL_FROM_ACTIVITY, mxConstants.ARROW_DIAMOND_THIN);
     expectModelContainsEdge('conditional_sequence_flow_from_gateway_id', SequenceFlowKind.CONDITIONAL_FROM_GATEWAY);

--- a/test/e2e/mxGraph.model.test.ts
+++ b/test/e2e/mxGraph.model.test.ts
@@ -273,13 +273,13 @@ describe('mxGraph model', () => {
     return cell;
   }
 
-  function expectModelContainsEdge(cellId: string, kind: SequenceFlowKind, startArrow?: string, expectedValue?: ExpectedFont): mxgraph.mxCell {
+  function expectModelContainsEdge(cellId: string, kind: SequenceFlowKind, startArrow?: string, expectedFont?: ExpectedFont): mxgraph.mxCell {
     const cell = expectModelContainsCell(cellId);
     expect(cell.style).toContain(kind);
 
     const state = bpmnVisu.graph.getView().getState(cell);
     expect(state.style[mxConstants.STYLE_STARTARROW]).toEqual(startArrow);
-    expectFont(state, expectedValue);
+    expectFont(state, expectedFont);
     return cell;
   }
 

--- a/test/e2e/mxGraph.model.test.ts
+++ b/test/e2e/mxGraph.model.test.ts
@@ -182,7 +182,9 @@ describe('mxGraph model', () => {
             <bpmndi:BPMNEdge bpmnElement="default_sequence_flow_id" id="E1373649849864_default_sequence_flow_id">
                 <di:waypoint x="342.0" y="351.0"/>
                 <di:waypoint x="390.0" y="351.0"/>
-                <bpmndi:BPMNLabel/>
+                <bpmndi:BPMNLabel labelStyle="LS1373649849858">
+                    <dc:Bounds height="12.804751171875008" width="94.93333333333335" x="153.67766754457273" y="371.3333333333333"/>
+                </bpmndi:BPMNLabel>
             </bpmndi:BPMNEdge>
             <bpmndi:BPMNEdge bpmnElement="normal_sequence_flow_id" id="E1373649849865_normal_sequence_flow_id">
                 <di:waypoint x="216.0" y="351.0"/>
@@ -271,12 +273,13 @@ describe('mxGraph model', () => {
     return cell;
   }
 
-  function expectModelContainsEdge(cellId: string, kind: SequenceFlowKind, startArrow?: string): mxgraph.mxCell {
+  function expectModelContainsEdge(cellId: string, kind: SequenceFlowKind, startArrow?: string, expectedValue?: ExpectedFont): mxgraph.mxCell {
     const cell = expectModelContainsCell(cellId);
     expect(cell.style).toContain(kind);
 
     const state = bpmnVisu.graph.getView().getState(cell);
     expect(state.style[mxConstants.STYLE_STARTARROW]).toEqual(startArrow);
+    expectFont(state, expectedValue);
     return cell;
   }
 
@@ -349,7 +352,7 @@ describe('mxGraph model', () => {
     expectModelContainsShape('inclusiveGateway_1', ShapeBpmnElementKind.GATEWAY_INCLUSIVE);
 
     // sequence flow
-    expectModelContainsEdge('default_sequence_flow_id', SequenceFlowKind.DEFAULT, MarkerConstant.ARROW_DASH);
+    expectModelContainsEdge('default_sequence_flow_id', SequenceFlowKind.DEFAULT, MarkerConstant.ARROW_DASH, expectedFont);
     expectModelContainsEdge('normal_sequence_flow_id', SequenceFlowKind.NORMAL);
     expectModelContainsEdge('conditional_sequence_flow_from_activity_id', SequenceFlowKind.CONDITIONAL_FROM_ACTIVITY, mxConstants.ARROW_DIAMOND_THIN);
     expectModelContainsEdge('conditional_sequence_flow_from_gateway_id', SequenceFlowKind.CONDITIONAL_FROM_GATEWAY);

--- a/test/unit/component/mxgraph/MxGraphRenderer.test.ts
+++ b/test/unit/component/mxgraph/MxGraphRenderer.test.ts
@@ -20,6 +20,9 @@ import ShapeBpmnElement from '../../../../src/model/bpmn/shape/ShapeBpmnElement'
 import { ShapeBpmnElementKind } from '../../../../src/model/bpmn/shape/ShapeBpmnElementKind';
 import Label, { Font } from '../../../../src/model/bpmn/Label';
 import { ExpectedFont } from '../parser/json/JsonTestUtils';
+import Edge from '../../../../src/model/bpmn/edge/Edge';
+import SequenceFlow from '../../../../src/model/bpmn/edge/SequenceFlow';
+import { SequenceFlowKind } from '../../../../src/model/bpmn/edge/SequenceFlowKind';
 
 function toFont(font: ExpectedFont): Font {
   return new Font(font.name, font.size, font.isBold, font.isItalic, font.isUnderline, font.isStrikeThrough);
@@ -31,6 +34,14 @@ function toFont(font: ExpectedFont): Font {
  */
 function newShapeBpmnElement(kind: ShapeBpmnElementKind): ShapeBpmnElement {
   return new ShapeBpmnElement('id', 'name', kind);
+}
+
+/**
+ * Returns a new `SequenceFlow` instance with arbitrary id and name.
+ * @param kind the `SequenceFlowKind` to set in the new `SequenceFlow` instance
+ */
+function newSequenceFlow(kind: SequenceFlowKind): SequenceFlow {
+  return new SequenceFlow('id', 'name', undefined, undefined, kind);
 }
 
 describe('mxgraph renderer', () => {
@@ -56,10 +67,23 @@ describe('mxgraph renderer', () => {
     expect(mxGraphRenderer.computeStyle(shape)).toEqual('intermediateCatchEvent;fontFamily=Arial;fontStyle=2');
   });
 
-  // it('compute edge style', () => {
-  //
-  //   const shape = new Shape('id', null, undefined, undefined);
-  //
-  //   mxGraphRenderer.computeStyle(null);
-  // });
+  it('compute style of edge with no label', () => {
+    const edge = new Edge('id', newSequenceFlow(SequenceFlowKind.CONDITIONAL_FROM_GATEWAY));
+    expect(mxGraphRenderer.computeStyle(edge)).toEqual('conditional_from_gateway');
+  });
+
+  it('compute style of edge with a no font label', () => {
+    const edge = new Edge('id', newSequenceFlow(SequenceFlowKind.NORMAL), undefined, new Label(undefined, undefined));
+    expect(mxGraphRenderer.computeStyle(edge)).toEqual('normal');
+  });
+
+  it('compute style of edge with label including strike-through font', () => {
+    const edge = new Edge('id', newSequenceFlow(SequenceFlowKind.CONDITIONAL_FROM_ACTIVITY), undefined, new Label(toFont({ size: 14.2, isStrikeThrough: true }), undefined));
+    expect(mxGraphRenderer.computeStyle(edge)).toEqual('conditional_from_activity;fontSize=14.2;fontStyle=8');
+  });
+
+  it('compute style of edge with label including underline font', () => {
+    const edge = new Edge('id', newSequenceFlow(SequenceFlowKind.DEFAULT), undefined, new Label(toFont({ isUnderline: true }), undefined));
+    expect(mxGraphRenderer.computeStyle(edge)).toEqual('default;fontStyle=4');
+  });
 });

--- a/test/unit/component/mxgraph/MxGraphRenderer.test.ts
+++ b/test/unit/component/mxgraph/MxGraphRenderer.test.ts
@@ -1,0 +1,73 @@
+/**
+ * Copyright 2020 Bonitasoft S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MxGraphRenderer from '../../../../src/component/mxgraph/MxGraphRenderer';
+import Shape from '../../../../src/model/bpmn/shape/Shape';
+import ShapeBpmnElement from '../../../../src/model/bpmn/shape/ShapeBpmnElement';
+import { ShapeBpmnElementKind } from '../../../../src/model/bpmn/shape/ShapeBpmnElementKind';
+import Label, { Font } from '../../../../src/model/bpmn/Label';
+import { ExpectedFont } from '../parser/json/JsonTestUtils';
+
+function toFont(font: ExpectedFont): Font {
+  return new Font(font.name, font.size, font.isBold, font.isItalic, font.isUnderline, font.isStrikeThrough);
+}
+
+/**
+ * Returns a new `ShapeBpmnElement` instance with arbitrary id and name.
+ * @param kind the `ShapeBpmnElementKind` to set in the new `ShapeBpmnElement` instance
+ */
+function newShapeBpmnElement(kind: ShapeBpmnElementKind): ShapeBpmnElement {
+  return new ShapeBpmnElement('id', 'name', kind);
+}
+
+describe('mxgraph renderer', () => {
+  const mxGraphRenderer = new MxGraphRenderer(null); // we don't care of mxgraph graph here
+
+  it('compute style of shape with no label', () => {
+    const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.TASK_USER));
+
+    const style = mxGraphRenderer.computeStyle(shape);
+    expect(style).toEqual('userTask');
+  });
+
+  it('compute style of shape with a no font label', () => {
+    const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.EVENT_END), undefined, new Label(undefined, undefined));
+
+    const style = mxGraphRenderer.computeStyle(shape);
+    expect(style).toEqual('endEvent');
+  });
+
+  it('compute style of shape with label including bold font', () => {
+    const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.GATEWAY_EXCLUSIVE), undefined, new Label(toFont({ name: 'Courier', size: 9, isBold: true }), undefined));
+
+    const style = mxGraphRenderer.computeStyle(shape);
+    expect(style).toEqual('exclusiveGateway;fontFamily=Courier;fontSize=9;fontStyle=1');
+  });
+
+  it('compute style of shape with label including italic font', () => {
+    const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH), undefined, new Label(toFont({ name: 'Arial', isItalic: true }), undefined));
+
+    const style = mxGraphRenderer.computeStyle(shape);
+    expect(style).toEqual('intermediateCatchEvent;fontFamily=Arial;fontStyle=2');
+  });
+
+  // it('compute edge style', () => {
+  //
+  //   const shape = new Shape('id', null, undefined, undefined);
+  //
+  //   mxGraphRenderer.computeStyle(null);
+  // });
+});

--- a/test/unit/component/mxgraph/MxGraphRenderer.test.ts
+++ b/test/unit/component/mxgraph/MxGraphRenderer.test.ts
@@ -38,30 +38,22 @@ describe('mxgraph renderer', () => {
 
   it('compute style of shape with no label', () => {
     const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.TASK_USER));
-
-    const style = mxGraphRenderer.computeStyle(shape);
-    expect(style).toEqual('userTask');
+    expect(mxGraphRenderer.computeStyle(shape)).toEqual('userTask');
   });
 
   it('compute style of shape with a no font label', () => {
     const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.EVENT_END), undefined, new Label(undefined, undefined));
-
-    const style = mxGraphRenderer.computeStyle(shape);
-    expect(style).toEqual('endEvent');
+    expect(mxGraphRenderer.computeStyle(shape)).toEqual('endEvent');
   });
 
   it('compute style of shape with label including bold font', () => {
     const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.GATEWAY_EXCLUSIVE), undefined, new Label(toFont({ name: 'Courier', size: 9, isBold: true }), undefined));
-
-    const style = mxGraphRenderer.computeStyle(shape);
-    expect(style).toEqual('exclusiveGateway;fontFamily=Courier;fontSize=9;fontStyle=1');
+    expect(mxGraphRenderer.computeStyle(shape)).toEqual('exclusiveGateway;fontFamily=Courier;fontSize=9;fontStyle=1');
   });
 
   it('compute style of shape with label including italic font', () => {
     const shape = new Shape('id', newShapeBpmnElement(ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH), undefined, new Label(toFont({ name: 'Arial', isItalic: true }), undefined));
-
-    const style = mxGraphRenderer.computeStyle(shape);
-    expect(style).toEqual('intermediateCatchEvent;fontFamily=Arial;fontStyle=2');
+    expect(mxGraphRenderer.computeStyle(shape)).toEqual('intermediateCatchEvent;fontFamily=Arial;fontStyle=2');
   });
 
   // it('compute edge style', () => {

